src/helpers.ts

```diff
@@ -974,19 +1106,21 @@ export function updateProtocolPosition(protocol: LendingProtocol, market: Market
   market.save();
 }
 
-function computeProportionIdle(market: Market): BigInt {
-  const offset = pow10(market._indexesOffset);
-  if (market._idleSupply && market._idleSupply!.gt(BigInt.zero())) {
-    const totalP2PSupplied = market._p2pSupplyAmount.times(market._p2pSupplyIndex).div(offset);
-    const proportionIdle = market._idleSupply!.times(offset).div(totalP2PSupplied);
-    if (proportionIdle.gt(offset)) return offset;
-    return proportionIdle;
-  }
-  return BigInt.zero();
-}
+export function updateP2PRates(event: ethereum.Event, market: Market, __MATHS__: IMaths): void {
```

Comment:
`updateP2PRates` can be called multiple times in a block, considers updating only the rates in this case. 

Later, we should have to handle the specific case of a donation to lenders through flash loans etc (linked with the bug of the previous week) 

Moreover the naming is a bit confusing, since youre updating both indexes and rates